### ISSUE TITLE
Bug: REST API payload format incorrect for TensorFlow/Keras models

### ISSUE THREAD

**Issue by rgasiorek** [rgasiorek/healthcare-ai-platform-databricks#8]
## Problem

Model serving endpoint returns HTTP 400 error when calling REST API:
```
{"error_code": "BAD_REQUEST", 
 "message": "Failed to enforce schema of data... 
 model signature requires tensor spec input, 
 the input pandas dataframe values should be scalar value..."}
```

## Root Cause

Used **wrong payload format** for TensorFlow/Keras models:

### What we used (WRONG):
```python
payload = {
    "dataframe_records": [{
        "image": img_array.tolist()
    }]
}
```

### What MLflow expects (CORRECT):
```python
payload = {
    "inputs": [img_array.tolist()]
}
```

## Why This Happens

MLflow infers model signature when logging:
```python
mlflow.keras.log_model(model, "model")
# Creates tensor signature: Tensor('float64', (-1, 64, 64, 3))
# Expects "inputs" format (tensor), not "dataframe_records" (pandas)
```

## Framework Differences

Different MLflow flavors expect different formats:

| Framework | Payload Format |
|-----------|----------------|
| TensorFlow/Keras | `{"inputs": [[...]]}` ✅ |
| Scikit-learn | `{"dataframe_records": [{...}]}` |
| PyTorch | `{"inputs": [[...]]}` |
| Custom pyfunc | Varies |

Our model uses **TensorFlow/Keras** → Must use `inputs` format\!

## Impact

- 4 out of 5 API requests failed in demo notebook
- Demo notebook couldn't show REST API predictions
- Pupils couldn't see comparison between SDK and REST API

## Solution

Changed payload format from `dataframe_records` to `inputs` in:
- Warm-up request
- Individual predictions
- Batch predictions
- Code examples in documentation

## Files Changed

- `notebooks/03_ml/demo_model_usage.py`: 3 occurrences fixed
- `notebooks/03_ml/deploy_serving_endpoint.py`: 3 occurrences fixed

## Resolution

✅ Fixed in commit: 3617ba4
✅ Deployed to Databricks
✅ Tested: REST API now accepts requests correctly

## Learning Opportunity

This is a **great teaching moment** for pupils about:
- MLflow model signatures
- Framework-specific serving formats
- Reading API error messages
- Debugging production ML systems

## Related

Part of Issue #6 - ML model POC implementation
